src/ocm/components/clusterConfiguration/manifestsConfiguration/components/CustomManifestsArray.tsx

```diff
@@ -0,0 +1,285 @@
+  showRemoveButton: boolean;
+  dataTestId: string;
+}) => (
+  <Button
+    aria-label="remove manifest"
+    style={{ visibility: showRemoveButton ? 'visible' : 'hidden' }}
+    variant="plain"
+  >
+    <MinusCircleIcon onClick={onRemove} data-testid={dataTestId} />
+  </Button>
+);
+
+const getManifestName = (manifestIdx: number) => `Custom manifest ${manifestIdx + 1}`;
+
+const SingleManifest = ({
+  fieldName,
+  manifestIdx,
+  onRemove,
+  onToggleExpand,
+  isExpanded,
+  isDisabled,
+  CollapsedCustomManifestComponent,
+  ExpandedCustomManifestComponent,
+  enableRemoveManifest,
+}: SingleManifestProps) => {
+  const [showRemoveButton, setShowRemoveButton] = React.useState(false);
+  const updateShowRemoveButton = (value: boolean) => {
+    if (isDisabled) {
+      return;
+    }
+    if (!enableRemoveManifest) {
+      setShowRemoveButton(false);
+    } else {
+      setShowRemoveButton(value);
+    }
+  };
+  const manifestFieldName = getFormikArrayItemFieldName(fieldName, manifestIdx);
+  return (
+    <Grid
+      onMouseEnter={() => updateShowRemoveButton(true)}
+      onMouseLeave={() => updateShowRemoveButton(false)}
+      hasGutter
+    >
+      <Flex>
+        <FlexItem>
+          <ExpandableSectionToggle
+            isExpanded={isExpanded}
+            onToggle={onToggleExpand}
+            direction="down"
+            data-testid={`toggle-manifest-${manifestIdx}`}
+          >
+            {getManifestName(manifestIdx)}
+          </ExpandableSectionToggle>
+        </FlexItem>
+        {!isDisabled && enableRemoveManifest && (
+          <FlexItem align={{ default: 'alignRight' }}>
+            <RemoveItemButton
+              onRemove={onRemove}
+              showRemoveButton={showRemoveButton}
+              dataTestId={`remove-manifest-${manifestIdx}`}
+            />
+          </FlexItem>
+        )}
+      </Flex>
+
+      {isExpanded ? (
+        <ExpandableSection isDetached key={manifestIdx} isExpanded>
+          <ExpandedCustomManifestComponent
+            fieldName={manifestFieldName}
+            manifestIdx={manifestIdx}
+            isDisabled={isDisabled}
+          />
+        </ExpandableSection>
+      ) : (
+        <CollapsedCustomManifestComponent
+          fieldName={manifestFieldName}
+          manifestIdx={manifestIdx}
+          isDisabled={isDisabled}
+        />
+      )}
+    </Grid>
+  );
+};
+
+type ExpandedManifests = { [manifestIdx: number]: boolean };
+
+const getExpandedManifestsInitialValue = (numManifests: number): ExpandedManifests => {
+  const ret = {};
+  for (let i = 0; i < numManifests; ++i) {
+    ret[i] = false;
+  }
+  return ret;
+};
+
+const getExpandedManifestsDefaultValue = (numManifests: number): ExpandedManifests => {
+  const ret = getExpandedManifestsInitialValue(numManifests);
+  if (numManifests === 1) {
+    ret[0] = true;
+  }
+  return ret;
+};
+
+const Manifests = <CustomManifestType,>({ push, remove, clusterId, ...props }: ManifestProps) => {
```

Comment:
Some notes here:

- I am able to continue with the initial empty manifest. Is that correct?

- If I don't touch the initial manifest and add a new one, I am not able to remove either until I set the file names and add content to both. Is that correct? Why am I not able to remove a manifest while any manifest in the array is invalid?

Reply:
You needs at least one manifest. You can't have an empty array.
I've added a validation to don't allow users to continue with an initial empty manifest.